test/unit/io/alignment/sam_tag_dictionary_test.cpp

```diff
@@ -0,0 +1,182 @@
+    EXPECT_TRUE((std::Same<sam_tag_type<"HI"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"IH"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"LB"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"MC"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"MD"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"MI"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"MQ"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"NH"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"NM"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"OC"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"OP"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"OQ"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"OX"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"PG"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"PQ"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"PT"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"PU"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"Q2"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"QT"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"QX"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"R2"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"RG"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"RT"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"RX"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"SA"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"SM"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"TC"_tag>::type, int32_t>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"U2"_tag>::type, std::string>));
+    EXPECT_TRUE((std::Same<sam_tag_type<"UQ"_tag>::type, int32_t>));
+}
+
+TEST(sam_tag_dictionary, set_function_known_tag)
+{
+    sam_tag_dictionary dict{};
+
+    dict.set<"NM"_tag>(3);
+    dict.set<"NM"_tag>(5); // overwrites previous
+    dict.set<"CO"_tag>("comment");
+    dict.set<"CG"_tag>(std::vector<int32_t>{3, 4, 5});
+
+    EXPECT_EQ(dict.count("NM"_tag), 1);
+    EXPECT_EQ(dict.count("CO"_tag), 1);
+    EXPECT_EQ(dict.count("CG"_tag), 1);
+
+    // access elements without using get
+    EXPECT_EQ(std::get<int32_t>(std::get<1>(*dict.find("NM"_tag))), 5);
+    EXPECT_EQ(std::get<std::string>(std::get<1>(*dict.find("CO"_tag))), "comment");
+    EXPECT_EQ(std::get<std::vector<int32_t>>(std::get<1>(*dict.find("CG"_tag))), (std::vector<int32_t>{3, 4, 5}));
+}
+
+TEST(sam_tag_dictionary, set_function_unknown_tag)
+{
+    sam_tag_dictionary dict{};
+
+    dict.set<"nm"_tag>('3');
```

Comment:
well the lower case `nm` tag is not defined by the SAM specs so I can assign whatever I want.. 
`dict.set<"NM"_tag>('3');` would in contrast not compile :point_up: 
If this is confusing I can change the char to `'a'`?

Reply:
If you intended a char, you can just keep it. I was just wondering whether the `''` are accidentally there ;)

Reply:
> If this is confusing I can change the char to

Indeed, the example is a little confusing, because the case-difference is not immediately visible. Also, are you sure this won't compile? Characters are implicitly convertible to numbers...

Reply:
Yes that's why I have the `char` before the `int32_t` in the variant type list so it will be chosen first (meta::in goes over the list sequentially from the start)